include/envoy/network/io_handle.h

```diff
@@ -74,6 +86,12 @@ template <typename T> struct IoHandleCallResult {
 
   virtual ~IoHandleCallResult() {}
 
+  IoHandleCallResult& operator=(IoHandleCallResult&& result) {
```

Comment:
Would prefer to not have any magic C++ generated for stuff like this. Can you add an explicit `move` method or just remove cases where this is needed?

Reply:
This is needed  almost everywhere a IoHandle interface call returns. So I think a move assignment is cleaner.